webapp/src/main/resources/public/js/components/branches/BranchesSearchResults.js

```diff
@@ -66,18 +72,46 @@ class BranchesSearchResults extends React.Component {
         );
     }
 
+    renderBranchTextUnitsPagination(id) {
+        return (
+            <Row>
+                <AltContainer store={BranchTextUnitsPaginatorStore}>
+                    <Paginator
+                        onPreviousPageClicked={() => {
+                            BranchesPageActions.changeSelectedBranchTextUnitIds([]);
+                            BranchTextUnitsPaginatorActions.goToPreviousPage();
+                            BranchTextUnitsParamActions.changeBranchStatisticId(id);
```

Comment:
I might be misunderstanding the full context but this seems odd to me.

Is this necessary? If the user is clicking back, does that not mean the id should not change?
You also set the id when the next is clicked as well. This leads me to think that maybe the id shouldn't change in these steps

Reply:
I have removed those lines